### ISSUE TITLE
[no-input-rename] proposal: alias should be allowed if input value is being transformed

### ISSUE THREAD

**Issue by adam-marshall** [angular-eslint/angular-eslint#2674]
### simplistic example:

parent component has a property `fontSize` which is a string-based enum, e.g. 'sm', 'lg', 'xl'

child component has an input `[fontSize]` to accept this incoming value.

but in the child component, this is not used as fontSize, it is transformed into a separate CSS class which dictates the size of a container, e.g. `fontSize => `container-${fontSize}`

so in the child component, if a developer sees a property `fontSize`, they might expect it's value to be one of the string-based enums, but actually it will be 'container-sm' not 'sm'.

```typescript
fontSize = input.required<string, FontSize>({ transform: fontSize => `container-${fontSize}` });
```

it isn't `fontSize` anymore.

## overcoming this

as Angular provides the `alias` feature and the `transform` solution, it could be handled like

```typescript
containerSize = input.required<string, FontSize>({ alias: 'fontSize', transform: fontSize => `container-${fontSize}` });
```

however this contravenes `@angular-eslint/no-input-rename`

the argument is: **as the meaning of the value has been transformed, if it retained the old input name, it would be even more confusing to the developer than looking at the component and wondering why it has an input [fontSize] but no property `fontSize`**

# alternative

it could be done with a `computed` but that introduces an extra line of code, and feels unnecessary - why not have the ability to `transform` an input and also not allow that transformation to change its meaning and use `alias` 

### proposal

perhaps an option for `@angular-eslint/no-input-rename`, to allow `alias` if `transform` is also used.

**Comment by JamesHenry** [comment]
@adam-marshall Thanks for the context, I think your reasoning for your specific instance is sound, but I don't feel like this is necessarily a generalizable enough case to warrant saying "whenever there is a transform, an alias is now appropriate", even if that is opt-in via an option.

The Angular docs actively discourage aliasing, as does the lint rule - but Angular still provides the _functionality_ to alias because there _are_ odd times where it makes the most sense. For these cases, an `eslint-disable` comment (with a brief inline explanation for maximum maintainability) is a totally appropriate course of action IMO. 

My personal opinion is that developers have an overly negative sentiment towards making exceptions in their own codebases for the sake of feeling "clean", but it often results in stronger overall linting setups because the rule isn't generically weakened by a dozen options changing the recommended common case (exaggerating here, not talking about this specific rule or case any more).
